dataset/table.py

```diff
@@ -116,7 +116,9 @@ def insert(self, row, ensure=None, types=None):
         Returns the inserted row's primary key.
         """
         row = self._sync_columns(row, ensure, types=types)
-        res = self.db.executable.execute(self.table.insert(row))
+        res = self.db.conn.execute(self.table.insert(), row)
+        # if not self.db.in_transaction:
```

Comment:
Shouldn't `db.commit()` be called here? Otherwise the transaction stays open (started by db.conn on first connect) until the user calls `db.commit()`.